### PR TITLE
Add Apache 2.0 License

### PR DESCRIPTION
**Pull request by mmqw**
/cc @rysharm @ajyang99 @howardyu97

### CODE REVIEW

**Review by gautamgupta — APPROVED** [review]
LGTM

File: LICENSE.md [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+      To apply the Apache License to your work, attach the following
+      boilerplate notice, with the fields enclosed by brackets "{}"
+      replaced with your own identifying information. (Don't include
+      the brackets!)  The text should be enclosed in the appropriate
+      comment syntax for the file format. We also recommend that a
+      file or class name and description of purpose be included on the
+      same "printed page" as the copyright notice for easier
+      identification within third-party archives.
+
+   Copyright {yyyy} {name of copyright owner}
```
Comment (GautamGupta) [inline LICENSE.md]: We should actually change this to `2017 UW Blueprint (uwblueprint.org)` 😄 

cc @bryanc208 for exact text

Comment (GautamGupta) [reply LICENSE.md]: Oh wait this should not have our info, my bad. Just read the text above.

Comment (rperryng) [inline LICENSE.md:189]: 😉

**Review by rperryng — COMMENTED** [review]
I think you missed something!

File: LICENSE.md (hunk 1 above)
Comment (gautamgupta) [reply LICENSE.md:189]: It's actually not supposed to be changed 😄 eg. https://github.com/apache/arrow/blob/master/LICENSE.txt#L190

Comment (rperryng) [reply LICENSE.md:189]: Did you read the paragraph above this line?

```
  To apply the Apache License to your work, attach the following
  boilerplate notice, with the fields enclosed by brackets "{}"
  replaced with your own identifying information. (Don't include
  the brackets!)  The text should be enclosed in the appropriate
  comment syntax for the file format. We also recommend that a
  file or class name and description of purpose be included on the
  same "printed page" as the copyright notice for easier
  identification within third-party archives.
```

**Pull request merged**